03-attribute-operations.qmd

```diff
@@ -337,24 +339,28 @@ This can be plotted as a map, as illustrated in @fig-join-coffee-production:
 #| label: fig-join-coffee-production
 #| fig-cap: 'World coffee production, thousand 60-kg bags by country, in 2017 (source: International Coffee Organization).'
 
-base = world.plot(color='white', edgecolor='lightgrey')
-world_coffee.plot(ax=base, column='coffee_production_2017');
+base = world_coffee.plot(color='white', edgecolor='lightgrey')
+coffee_map = world_coffee.plot(ax=base, column='coffee_production_2017')
+coffee_map.set_title('Coffee production')
 ```
 
-For joining to work, a 'key variable' must be supplied in both datasets. 
-In this case, both `world_coffee` and world objects contained a variable called `name_long`.
-By default `pd.merge` uses all variables with matching names. 
-However, it is recommended to explicitly specify the names of the columns to be used for matching, like we did in the last example. 
+To work, attribute-based joins need a 'key variable' in both datasets (`on` parameter of `pd.merge`). 
+In the above example, both `world_coffee` and `world` DataFrames contained a column called `name_long`.
+(By default `pd.merge` uses all columns with matching names. However, it is recommended to explicitly specify the names of the columns to be used for matching, like we did in the last example.)
 
-In case where variable names are not the same, you need to rename (using `.rename`) them so that it is, then proceed with the join.
+In case where column names are not the same, you can use `left_on` and `right_on` to specify the respective columns.
```

Comment:
I usually do the former, but this is subjective, and I agree that the latter is a more popular choice

Reply:
I'm fine with mentioning both options but the original "you need to rename" statement may be misleading, i.e. it sounds like there is no other way

Reply:
You're right! "Need to" was incorrect. Then perhaps we can say: 
either (1) rename and use `on`, or (2) use `left_on`+`right_on`.